src/tests/ftest/harness/advanced.py

```diff
@@ -53,6 +54,10 @@ def test_core_files(self):
         if len(result) > 1 or result[0]["exit_status"] != 0:
             self.fail("Error sending a signal 6 to {} on {}".format(pid, host))
 
+        # Wait for the core dump to finish
+        self.log.info("Sleeping 30s to allow the dump to finish")
+        sleep(30)
```

Comment:
I am sure you realize this but of course this just helps this one test of generating stacktraces and does not help the general problem of all of the other tests that may cause a segfault having their engines/control-plane processes killed too quickly when they cause a segfault and are dumping a big core file.

So while this will ameliorate this one test from creating a truncated core file, the more general problem still exists.

Reply:
I've resolved the issue where processing an incomplete core file was causing the launch.py to exit prematurely, so I think we can address the copying of an incomplete core file in a separate PR as it may be more of a harness issue in terms of waiting for the core file to be complete before moving on from the test.